src/events/installationStateChangedEmitter.ts

```diff
@@ -0,0 +1,26 @@
+import { Event, EventEmitter } from "vscode";
+
+export enum InstallationState {
+  Undefined,
+  ExistingInstallation,
+  NewInstallation,
+}
+
+export class InstallationStateEmitter {
+  private static emitter = new EventEmitter<InstallationState>();
```

Comment:
remove all the static, and export an instance of this class

Reply:
why is that better? the goal is to have a singleton